System/Posix/Process/Common.hsc

```diff
@@ -449,6 +449,10 @@ The exception masking state of the executed action is inherited
 threads are not copied into the child process, it's easy to go wrong:
 e.g. by accessing some shared resource that was held by another thread
 in the parent.
+
+GHC note: 'forkProcess' is not currently very well supported when using multiple
+processors (@+RTS -N@), although it is supported with @-threaded@ as
+long as only one processor is being used.
```

Comment:
I think `processor` is the wrong aspect of the issue.  A program can be multi-threaded even on a single CPU.  This is really is about (OS) threads. If the RTS is running in a single-threaded process, all is well.  In multi-threaded processes, fork(2) and friends have poor semantics. A better solution to the OP's problem would be `posix_spawn`.

Reply:
I just copy pasted the original commit from unix: https://github.com/haskell/unix/commit/05eea1ea715745d4e2086d3b25a14f35f424045c

What exact wording do you propose?

Reply:
I would replace processor with either __OS thread__ or __capability__